.github/workflows/chart.yaml

```diff
@@ -14,22 +14,34 @@ jobs:
       - name: Checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
 
-      - name: Helm lint
-        working-directory: ./cost-analyzer
-        run: helm lint
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@b43128a8b25298e1e7b043b78ea6613844e079b1 # v2.6.0
+
+      # Lint all chart values including those in the ci directory.
+      - name: Run chart-testing (lint)
+        run: ct lint --chart-dirs=cost-analyzer/ --charts cost-analyzer/ --validate-maintainers=false
 
+      # Run `helm template` on the main values file plus all those in the ci directory.
       - name: Helm template
-        working-directory: ./cost-analyzer
-        run: helm template . --set global.prometheus.enabled=false --set global.grafana.enabled=false > full.yaml
+        run: |
+          helm template cost-analyzer/ -f cost-analyzer/values.yaml > full.yaml
+          directory="cost-analyzer/ci"
+          for file in "$directory"/*; do
+          if [ -f "$file" ]; then
+              helm template cost-analyzer/ -f "$file" >> full.yaml
```

Comment:
If we are looping over the entire `cost-analzyer/ci` directory. Then rendering the templates and appending output to `full.yaml`

Is the expected behavior for `Kubeconform` to later fail if at least one of these `values.yaml` files have rendered incorrectly?

Reply:
No, Kubeconform is only used to assess conformance of the final, rendered output against Kubernetes resource specs. For example, let's say you had `spec.foo: bar` in a Pod. While this may template correctly, it's not conformant to the Pod specification. A `helm template` would not catch this because the templating may occur successfully, but the final resource does not conform to the known Pod spec. Kubeconform is used to ensure that the (successfully) templated output are actually valid K8s resources.